pandas/tests/arrays/test_boolean.py

```diff
@@ -251,6 +251,13 @@ def test_coerce_to_numpy_array():
         np.array(arr, dtype="bool")
 
 
+def test_to_boolean_array_from_strings(na_value):
+    result = BooleanArray._from_sequence_of_strings(["True", "False"])
```

Comment:
```suggestion
    result = BooleanArray._from_sequence_of_strings(np.array(["True", "False", np.nan], dtype=object))
```

(that's the type of object that it receives from the parser)